File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => [0, 2, 5, 3, 7, 8, 4];

    public int Today() => birdsPerDay[^1];

    public void IncrementTodaysCount() => birdsPerDay[^1]++;

    public bool HasDayWithoutBirds()
    {
        foreach (int day in birdsPerDay){
            if (day == 0) return true;
        }
        return false;
    }

    public int CountForFirstDays(int numberOfDays)
    {
        int birdsCounter = 0;
        for (int i = 0; i < numberOfDays; i++){
            birdsCounter += birdsPerDay[i];
        }
        return birdsCounter;
    }

    public int BusyDays()
    {
        int busyDaysCounter = 0;
        foreach (int day in birdsPerDay){
            if (day >= 5) busyDaysCounter++;
        }
        return busyDaysCounter;
    }
}

File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public static int[] getLastWeek() {
        final int[] birdsPerDayLastWeek = {0, 2, 5, 3, 7, 8, 4};
        return birdsPerDayLastWeek;
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1] += 1;
    }

    public boolean hasDayWithoutBirds() {
        for (int i = 0; i < birdsPerDay.length; i++){
            if (birdsPerDay[i] == 0){
                return true;
            }
        }
        return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
        int counter = 0;
        
        for (int i = 0; i < (numberOfDays > 7? 7: numberOfDays); i++){
            counter += birdsPerDay[i];
        }
        return counter;
    }

    public int getBusyDays() {
        int counter = 0;
        for (int i = 0; i < birdsPerDay.length; i++){
            if (birdsPerDay[i] >= 5){
                counter++;
            }
        }
        return counter;
    }
}
